crypto/ocsp/ocsp_http.c

```diff
@@ -13,29 +13,35 @@
 
 #ifndef OPENSSL_NO_OCSP
 
-# ifndef OPENSSL_NO_DEPRECATED_3_0
 int OCSP_REQ_CTX_set1_req(OSSL_HTTP_REQ_CTX *rctx, const OCSP_REQUEST *req)
 {
     return OSSL_HTTP_REQ_CTX_i2d(rctx, "application/ocsp-request",
                                  ASN1_ITEM_rptr(OCSP_REQUEST),
                                  (ASN1_VALUE *)req);
 }
-# endif
 
 OSSL_HTTP_REQ_CTX *OCSP_sendreq_new(BIO *io, const char *path,
-                                    OCSP_REQUEST *req, int maxline)
+                                    const OCSP_REQUEST *req, int maxline)
 {
-    BIO *req_mem = HTTP_asn1_item2bio(ASN1_ITEM_rptr(OCSP_REQUEST),
-                                      (ASN1_VALUE *)req);
-    OSSL_HTTP_REQ_CTX *res =
-        HTTP_REQ_CTX_new(io, io, 0 /* no HTTP proxy used */, NULL, NULL, path,
-                         NULL /* headers */, "application/ocsp-request",
-                         req_mem /* may be NULL */,
-                         maxline, 0 /* default max_resp_len */,
-                         0 /* no timeout, blocking indefinite */, NULL,
-                         1 /* expect_asn1 */);
-    BIO_free(req_mem);
-    return res;
+    OSSL_HTTP_REQ_CTX *rctx = NULL;
+
+    if ((rctx = OSSL_HTTP_REQ_CTX_new(io, io, 1 /* POST */,
+                                      maxline, 0 /* default max_resp_len */,
+                                      0 /* no timeout, blocking indefinitely */,
+                                      NULL, 1 /* expect_asn1 */)) == NULL)
+        return NULL;
+
+    if (!OSSL_HTTP_REQ_CTX_set_request_line(rctx, NULL, NULL, path))
+        goto err;
+
+    if (req != NULL && !OCSP_REQ_CTX_set1_req(rctx, req))
+        goto err;
```

Comment:
I see no reason to keep `OCSP_REQ_CTX_set1_req`, as it remains the only function with the `OSSL_REQ_CTX` prefix.
How about just transferring the `OSSL_HTTP_REQ_CTX_i2d` call here?

Reply:
> I see no reason to keep `OCSP_REQ_CTX_set1_req`, as it remains the only function with the `OSSL_REQ_CTX` prefix.

I've now replaced also  `OCSP_REQ_CTX_set1_req` by a macro that is enabled only if OPENSSL_NO_DEPRECATED_3_0 is not defined,
as was done already for the other deprecated functions here.

> How about just transferring the `OSSL_HTTP_REQ_CTX_i2d` call here?

All right, did so though this introduced minor redundancy.